gateway-service-admin/src/main/java/org/apache/knox/gateway/service/admin/AliasResource.java

```diff
@@ -100,15 +105,20 @@ public Response postAlias(@PathParam("topology") final String topology,
     final AliasService as = getAliasService();
 
     try {
+      /* check if the string is properly encoded and expect encoded values */
+      topology = URLDecoder.decode(topology, StandardCharsets.UTF_8.name());
+      alias = URLDecoder.decode(alias, StandardCharsets.UTF_8.name());
+      value = URLDecoder.decode(value, StandardCharsets.UTF_8.name());
+
       as.addAliasForCluster(topology, alias, value);
       return status(Response.Status.CREATED).
-          entity("{ \"created\" : { \"topology\": \"" + topology
-              + "\", \"alias\": \"" + alias + "\" } }").
+          entity("{ \"created\" : { \"topology\": \"" + Encode.forHtml(topology)
+              + "\", \"alias\": \"" + Encode.forHtml(alias) + "\" } }").
           type(MediaType.APPLICATION_JSON_TYPE).
           build();
-    } catch (AliasServiceException e) {
+    } catch (AliasServiceException | UnsupportedEncodingException e) {
       return status(INTERNAL_SERVER_ERROR).
-          entity("Error adding alias " + alias + " for cluster " + topology
+          entity("Error adding alias " + Encode.forHtml(alias) + " for cluster " + Encode.forHtml(topology)
               + ", reason: " + e.toString()).
```

Comment:
Will the e.toString() potentially print the value of the alias from line 111 on a decoding error?

Reply:
Interesting, there is a chance it could, I will remove it.

Reply:
I think if the URLDecoder stuff is removed then there shouldn't be a reason for the alias to be printed. Not sure though.